### DISCUSSION TITLE
Releasing NTS 2.6.0

### DISCUSSION THREAD

**Discussion by airbreather** [NetTopologySuite/NetTopologySuite#739] [General]
Some folks from across my company have pinged me about the coverage tools that have been ported to NTS since the 2.5.0 release, very enthusiastic to get their hands on this stuff since this has been a pain point at our company for decades.

My immediate response was to point out that NTS stable releases almost always track JTS stable releases: since there has been no stable JTS release with these APIs, then it would be a bit of a risk to jump the gun on this, since we could no longer easily adopt breaking changes that JTS might easily make between now and when the APIs land in a stable release.

However, the more I explore it, the less satisfied I am with that answer. Here's a timeline:

- **2022-06-21**: JTS 1.19.0 is released
- **2022-06-22** (*1 day later*): NTS 2.5.0 is released
- **2022-07-24** (*1 month later*): Martin makes [this post](https://lin-ear-th-inking.blogspot.com/2022/07/polygonal-coverages-and-operations-in.html) about polygon coverages
- **2022-07-28** (*4 days later*): locationtech/jts#900 is created
- **2022-07-28** (*1 hour later*): that PR is merged
- **2022-08-04** (*1 week later*): NetTopologySuite/NetTopologySuite@a6388051 ports the changes from that PR into NTS
- **2022-08-05** (*1 day later*): libgeos/geos@62c928c9 ports the changes from that PR into GEOS
- **2023-06-27** (*~11 months later*): [GEOS 3.12.0](https://github.com/libgeos/geos/releases/tag/3.12.0) is released, the first non-beta with the new coverage APIs
- **2023-11-11** (*~4 months later*): [GEOS 3.12.1](https://github.com/libgeos/geos/releases/tag/3.12.1) is released, the current latest release
- **2024-01-17** / **2024-01-18** (*~10 weeks later*): Jody and Martin have an exchange over here about a new JTS release https://github.com/locationtech/jts/issues/996#issuecomment-1896273853
- **2024-03-06** (*~7 weeks later*): I write this timeline

To sum that up, the new polygon coverage APIs have been available to very early adopters for almost as long as stable-channel users have had JTS 1.19, and more generally to GEOS users (without a "beta" tag) for like 8 months now.

@FObermaier I'm curious what you might think about a 2.6.0 release? If you want to see JTS stabilize first, then I will support that decision entirely, but I think it would still be nice to at least publish a 2.6.0 prerelease on NuGet to make it easier for more early adopters to get their hands on this.

I've been involved in some internal projects at my company where it made more sense to use NTS 2.6.0 from the prerelease feed than any alternative at hand, and I am quite happy with that decision, even after seeing a couple of questionable-looking outputs from the coverage validator that I have been procrastinating to do the work to isolate and report (just two weird corner cases, pun intended, that look unrealistic but not definitely obviously illegal).

**Comment by FObermaier** [comment]
@airbreather, in general I have no objections to drive a different release schedule than JTS. In this particular situation though, I think we should wait for the next JTS release, since that is due.
